src/main/java/seedu/address/model/delivery/SupplierIndex.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.model.delivery;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+/**
+ * Represents the position of the supplier inside the addressbook that is involved in the delivery.
+ */
+public class SupplierIndex {
+    public static final String MESSAGE_CONSTRAINTS =
+            "SupplierIndex should be a positive number greater than 0"
+                    + " and smaller than total number of suppliers";
+    public final int index;
+
+    /**
+     * Constructs a {@code SupplierIndex}.
+     *
+     * @param index A index string.
+     */
+    public SupplierIndex(String index) {
+        requireNonNull(index);
+        checkArgument(isValidIndex(index), MESSAGE_CONSTRAINTS);
+        this.index = Integer.parseInt(index);
+    }
+
+    /**
+     * Returns true if the given string represents a positive-ingeger.
+     *
+     * @param test String value of Supplier_index provided by user.
+     * @return True if test is an nteger and false otherwise.
+     */
+    public static boolean isValidIndex(String test) {
+        try {
+            Integer.parseInt(test);
+        } catch (NumberFormatException e) {
+            return false;
+        }
+        return true;
+    }
+    @Override
+    public String toString() {
+        return "" + index;
```

Comment:
i think this toString should just return the value of the index? Then we can have a separate displayFormatter method to return <int> index instead. What do you think about this Jian Zhi?